### ISSUE TITLE
Mark class as invalid when parent classes/interfaces cannot be located

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#511]
In `Psalm\Codebase\Populator` methods `populateDataFromParentClass`, `populateInterfaceDataFromParentInterfaces`  and `populateDataFromImplementedInterfaces` we should mark a class as invalid if one or more of the classes/interfaces it depends on doesn't exist.

**Comment by weirdan** [comment]
This causes problems when classes (interfaces in my case) are created with `class_alias()`. We use `zend-expressive` here, which depends on `webimpress/http-middleware-compatibility` (`expressive` is not compatible with PSR-15 yet). 
This compatibility package is just a proxy that aliases the available implementation of `http-interop` interfaces into it's own namespace, so you could depend on any version (see https://github.com/webimpress/http-middleware-compatibility/blob/master/autoload/http-middleware.php )

**Comment by muglug** [comment]
@weirdan I've added https://github.com/vimeo/psalm/issues/514 to track this – I think an explicit flag is probably necessary. Or is your usage conditional on certain parameters?

**Comment by weirdan** [comment]
I unconditionally use `Zend\Expressive\Application` that depend on `webimpress/compat` interfaces that are kinda polymorphic (e.g. they change depending on what version of `http-interop` is installed). By "change" I mean the same interface could have different method names and signatures, conditional on the http-interop version because, under the hood, they are actually different interfaces). 

This must be hell for static analysis.

**Comment by muglug** [comment]
But for the purposes of Psalm's analysis, it's ok to use just one of the two?

**Comment by weirdan** [comment]
Of the three, and that's probably not gonna work. Imagine this:
```php
if (rand(0,1)) {
   interface AncientMiddleware { public function next(); }
   define('METHOD', 'next');
} elseif (rand(0,1)) {
   interface OldMiddleware { public function process(); }
   define('METHOD', 'next');
} else {
   interface NewMiddleware { public function handle(); }
   define('METHOD', 'handle');
}
```
Then, depending on what's available you're aliasing one of those to `MiddlewareInterface`. If that was done through inheritance, one would have to declare this `MiddlewareInterface` as empty marker interface and make all three extend that. But if psalm would think `MiddlewareInterface` is, say, `AncientMiddleware` it would not complain about calls to `MiddlewareInterface::next()`, even though as a client of this compat interface you should only be calling `MiddlewareInterface::{METHOD}()`

Long story short, I don't think this is worth the hassle, and has great potential for screwing up analysis badly. So I just need to suppress those errors.

**Comment by muglug** [comment]
Oh. That's some code right there.

**Comment by muglug** [comment]
> So I just need to suppress those errors.

I've changed the issue type to `MissingDependency`, so hopefully it'll be easier to suppress
